crates/http/src/lib.rs

```diff
@@ -391,6 +400,9 @@ pub(crate) fn compute_default_headers<'a>(
 /// All HTTP executors must implement this trait.
 #[async_trait]
 pub(crate) trait HttpExecutor: Clone + Send + Sync + 'static {
+    // TODO: allowing this lint because I want to gather feedback before
+    // investing time in reorganising this
+    #[allow(clippy::too_many_arguments)]
```

Comment:
We ran into this before and punted by...removing an unnecessary argument. :neutral_face:

I'm fine leaving this for now since these `HttpExecutor`s are ripe for a broader refactoring. Should probably reword the comment though.